pkg/kubelet/kubelet.go

```diff
@@ -1689,6 +1691,18 @@ func (kl *Kubelet) syncPod(o syncPodOptions) error {
 		}
 	}
 
+	// If pod is not host network, check network-readiness (only fails the cases with explicit not true value)
```

Comment:
just to clarify the case when this annotation is needed. in line 1592, the runtimeState.NetworkError() has check for network related errors reported from runtime. is this annotation covers some other cases where runtimeState network is ready but somehow it is not ready for a particular pod under some situations ?

Reply:
runtimeState.NetworkError() is about node's status (mostly about cni). network-readiness annotation is about the pod only; some external network provider may need this to sync with arktos (kubelet in this case), in order to prevent premature  pod launch before the network provider finishes _mandatory_ preparation (if there is any)